File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/model/ConversationContext.java
package com.study.webflux.rag.domain.dialogue.model;

import java.util.List;

public record ConversationContext(
	List<com.study.webflux.rag.domain.dialogue.model.ConversationTurn> turns
) {
	public ConversationContext {
		if (turns == null) {
			turns = List.of();
		}
	}

	public boolean isEmpty() {
		return turns.isEmpty();
	}

	public int size() {
		return turns.size();
	}

	public static ConversationContext empty() {
		return new ConversationContext(List.of());
	}

	public static ConversationContext of(
		List<com.study.webflux.rag.domain.dialogue.model.ConversationTurn> turns) {
		return new ConversationContext(turns);
	}
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/dialogue/model/ConversationContext.kt
package com.miyou.app.domain.dialogue.model

data class ConversationContext(
    val turns: List<ConversationTurn> = emptyList(),
) {
    fun isEmpty(): Boolean = turns.isEmpty()

    fun size(): Int = turns.size

    companion object {
        fun empty(): ConversationContext = ConversationContext(emptyList())

        fun of(turns: List<ConversationTurn>): ConversationContext = ConversationContext(turns)
    }

    fun turns(): List<ConversationTurn> = turns
}
